core/src/main/java/org/infinispan/interceptors/impl/InvalidationCacheWriterInterceptor.java

```diff
@@ -0,0 +1,23 @@
+package org.infinispan.interceptors.impl;
+
+import org.infinispan.commands.write.RemoveCommand;
+import org.infinispan.configuration.cache.CacheType;
+import org.infinispan.context.InvocationContext;
+
+/**
+ * Extends the {@link CacheWriterInterceptor} to add logic for {@link CacheType#INVALIDATION}.
+ * <p>
+ * With invalidation, the {@link RemoveCommand} should always try to remove from persistence, even if the originator
+ * does not have the data in memory. The removing non-existing value logic has an optimization that skips the
+ * persistence removal; it is disabled in this interceptor.
+ *
+ * @since 16.0
+ */
+public class InvalidationCacheWriterInterceptor extends CacheWriterInterceptor {
```

Comment:
Just food for thought: Is this an issue if someone has invalidation cache with passivation..? Although, I think that _shouldn't_ be possible as passivation is not supported with shared stores and invalidation cache really needs a shared store.

Reply:
Although invalidation requires a shared store it doesn't make as much sense to be configured as such I guess. I thisthat we may want to have a validation check to prevent passivation with INVALIDATION caches.

Reply:
As you mentioned, shared + passivation is not possible and it will throw an exception.

`invalidation` + `non-shared-store` may be (is!) broken. The `CacheWriterInterceptor` and subclasses ignore the `InvalidateCommand`, so it will never remove anything from the disk.